codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -178,4 +199,81 @@ private void writeAwsConfigConstructor(Model model, ServiceShape service, GoWrit
                 });
         writer.write("");
     }
+
+    private static class AwsConfigField extends ConfigField {
+        private final boolean generatedOnClient;
+        private final BiPredicate<Model, ServiceShape> servicePredicate;
+        private final Symbol resolveFunction;
+
+        private AwsConfigField(Builder builder) {
+            super(builder);
+            this.generatedOnClient = builder.generatedOnClient;
+            this.servicePredicate = builder.servicePredicate;
+            this.resolveFunction = builder.resolveFunction;
+        }
+
+        public boolean isGeneratedOnClient() {
+            return generatedOnClient;
+        }
+
+        public Optional<BiPredicate<Model, ServiceShape>> getServicePredicate() {
+            return Optional.ofNullable(servicePredicate);
+        }
+
+        public Optional<Symbol> getResolverFunction() {
+            return Optional.ofNullable(resolveFunction);
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        private static class Builder extends ConfigField.Builder {
+            private boolean generatedOnClient;
```

Comment:
Probably best to just default this to `true` since that's the actual value almost always